src/xtgeo/grid3d/_grid_import_xtgcpgeom.py

```diff
@@ -1,19 +1,68 @@
-# coding: utf-8
 """Private module, Grid Import private functions for xtgeo based formats."""
+from __future__ import annotations
 
 import json
 from struct import unpack
+from typing import TYPE_CHECKING, Literal, Tuple, TypedDict, Union, overload
 
 import h5py
 import numpy as np
+from typing_extensions import NotRequired, TypeAlias
 
 import xtgeo.common.sys as xsys
 from xtgeo.common import null_logger
+from xtgeo.grid3d._gridprop_import_xtgcpprop import _read_filelike
 
 logger = null_logger(__name__)
 
+if TYPE_CHECKING:
+    from xtgeo.common.sys import _XTGeoFile
 
-def convert_subgrids(sdict):
+LMIN: TypeAlias = Literal["min"]
+LMAX: TypeAlias = Literal["max"]
+IJKRANGE: TypeAlias = Tuple[
+    Union[int, LMIN],
+    Union[int, LMAX],
+    Union[int, LMIN],
+    Union[int, LMAX],
+    Union[int, LMIN],
+    Union[int, LMAX],
+]
+
+
+class _RESULT(TypedDict):
+    coordsv: np.ndarray
+    zcornsv: np.ndarray
+    actnumsv: np.ndarray
+    subgrids: NotRequired[dict[str, range]]
+
+
+class __REQUIRED_(TypedDict):
+    xshift: np.ndarray
+    xscale: np.ndarray
+    yshift: np.ndarray
+    yscale: np.ndarray
+    zshift: np.ndarray
+    zscale: np.ndarray
+    subgrids: NotRequired[dict[str, int]]
+
+
+class _META(TypedDict):
+    _required_: __REQUIRED_
+
+
+@overload
+def convert_subgrids(sdict: dict[str, int]) -> dict[str, range]:
+    ...
+
+
+@overload
+def convert_subgrids(sdict: None) -> None:
+    ...
+
+
+# flake8: noqa: F841
```

Comment:
Maybe put the `noqa: F841` on the line it's needed rather than over the whole file? This won't just cover this function